packages/shortest/README.md

```diff
@@ -73,6 +79,42 @@ shortest('Login to the app using username and password', {
 });
 ```
 
+## Test Dependencies
+You can create reusable test steps and chain them together. This helps organize complex test flows and maintain state between related tests:
+
+```typescript
+import { shortest, after } from '@antiwork/shortest';
+
+// Define reusable test steps
+const login = shortest('User can login', { 
+  email: 'test@example.com',
+  password: 'password'
+});
+
+const settings = shortest('Click settings button');
+
+// Chain tests with dependencies
+after(login, settings)
+  .shortest('Change language to Arabic')
+  .expect('Language should be Arabic');
+
+// Store intermediate steps
+const changeLanguage = after(login, settings)
+  .shortest('Change language to Arabic');
+
+// Use stored steps in other tests
+after(changeLanguage)
+  .shortest('Verify Arabic UI')
+  .expect('UI elements should be in Arabic');
+```
```

Comment:
```suggestion
// Use stored steps in other tests
shortest(login).shortest(changeLanguage).shortest(changeLanguageBack)
```

What about something like this? Keeps it all as shortest and keeps it very clean and simple.

Reply:
This may mean that writing tests could be as simple as:

```
shortest(login).shortest(changeLanguage).shortest(changeLanguageBack)
```

```
shortest(login).shortest(changeLanguageToJapanese).shortest(changeLanguageBackToEnglish)
```

And AI could automatically "cache" the login step?

(This gets a little crazy so maybe in another PR. haha)

Reply:
man, love that, it sounds great, but I agree with the last! let's make it seperated that this

Reply:
this approach is great but I don't think it will replace this approach of `after()` function since we can depend on one shortest test in mutliple `after()`

```typescript
const login = shortest("login ...", {...credentials})

after(login).shortest("The dark mode could be changed")
after(login).shortest("The language should be english")
```

Reply:
ofc we can develop it in that approach too if we checked the type of the input if it's ChainTest or string.

Reply:
Yeah think we should detect!